### PR TITLE
`Lectures`: Fix the order of lecture units when merging pdf files

### LINKED ISSUE #6645: `Lecture PDF slides`: they are not sorted when you download the complete PDF
### Describe the bug

When you create a lecture with UNITS with several PDF files, the button download complete PDF creates a unique PDF but is sorted according to the time when the units were added. If you change the order of the units. This does not affect in the creation of the complete PDF. 

### To Reproduce

1. Go to a course
2. Create a Lecture
3. Create 2 units (at least) and upload the corresponding PDF 
4. Verify as a student that the button works `create complete PDF` works and create a unique PDF in the order of the units. 
5. As instructor change the order of the Units
6. Verify the order of the new PDF does not change. 

### Expected behavior

If an instructor changes the order of the units, the button `create complete PDF` should create the PDF with the new order of the units.  

### Screenshots

_No response_

### Which version of Artemis are you seeing the problem on?

6.2.0

### What browsers are you seeing the problem on?

Chrome

### Additional context

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context
Closes #6645

### Description
In #4972 I implemented that in the merged pdf of all attachment units the pages should follow the order of the units.

Later #5099 fixed a race-condition by adding `@Transient` to the attribute, which broke this feature. Now, hibernate ignores the saved value when building the object, its value is always 0.

To combine both PRs I now completely removed the individual order in the attachment unit. It still gets saved in the database and the list used in the lecture follows this order. When merging the PDF file, we now retrieve the a ordered list directly from the database.

I was not able to retrieve the order without using the lecture in the query. This is the most performance-optimal query I was able to write here, if you have any other ideas on how this could be improved please let me know. You can find more information on the used INDEX and TREAT functions here: https://en.wikibooks.org/wiki/Java_Persistence/JPQL#Special_Operators

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Student
- 1 Lecture with multiple .pdf attachment units

1. Student: Download the merged pdf file for the lecture
2. Instructor: Re-order the units.
3. Student: Download the merged pdf again, verify the changed order
4. Repeat that process a few times and check that race conditions as in #5099 are not noticable

#### Performance Review
- [x] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### CODE REVIEW

**Review by pellumbbaboci — APPROVED** [review]
Tested during testing session on test4, it works as expected.

**Review by rriyaldhi — APPROVED** [review]
Tested on TS4 during testing session and works as described.

**Review by rriyaldhi — APPROVED** [review]
Reapprove after test fix.

**Review by lennart-keller — APPROVED** [review]
Testen on ts4 during testing session. Works really well.

File: src/test/java/de/tum/in/www1/artemis/FileIntegrationTest.java [hunk 1]
```diff
@@ -359,23 +360,66 @@ void testGetLecturePdfAttachmentsMerged() throws Exception {
...
+        lecture.getLectureUnits().add(0, unit3);
+        lectureRepo.save(lecture);
+
+        // The test setup needs at least TA right for creating a lecture with files.
+        database.changeUser(TEST_PREFIX + "student1");
+
+        try (PDDocument mergedDoc = retrieveMergeResult(lecture)) {
+            assertThat(mergedDoc.getNumberOfPages()).isEqualTo(5);
+            PDPage firstPage = mergedDoc.getPage(0);
+            assertThat(firstPage.getCropBox().getHeight()).isEqualTo(4);
```
Comment (julian-christl) [inline src/test/java/de/tum/in/www1/artemis/FileIntegrationTest.java:408]: Please add a comment how this verifies the correct order. That's unclear.

**Review by julian-christl — APPROVED** [review]
Looks good, thanks for the changes

**Review by julian-christl — APPROVED** [review]
Looks good, thanks for the changes

**Review by tobias-lippert — APPROVED** [review]
Repprove, only one comment in a test file was added

**Review by lennart-keller — APPROVED** [review]
Reapprove. Code lgtm, too.

**Review by rriyaldhi — APPROVED** [review]
Reapprove

**Pull request merged**